citrination_client/models/service_status.py

```diff
@@ -0,0 +1,115 @@
+from citrination_client.models import Event
+
+class ServiceStatus(object):
+    """
+    The status of a service for a given model on Citrination. The status of a
+    service is summarized by the ready value indicating whether or not the service
+    is in a state where it can be utilized. It also includes more detailed
+    information about the current state of the service in the form of a descriptive
+    sentence, a context string and an event object.
+    """
+
+    @staticmethod
+    def from_response_dict(response_dict):
+        if "event" in response_dict:
+            event_dict = response_dict["event"]
+            event = Event(
+                title = event_dict["title"],
+                subtitle = event_dict["subtitle"],
+                normalized_progress = event_dict["normalizedProgress"]
+            )
+
+            if "subevent" in response_dict["event"]:
+                subevent_dict = response_dict["event"]["subevent"]
+                subevent = Event(
+                    title = subevent_dict["title"],
+                    subtitle = subevent_dict["subtitle"],
+                    normalized_progress = subevent_dict["normalizedProgress"]
+                )
+                event.subevent = subevent
+        else:
+            event = None
+
+        return ServiceStatus(
+            ready = response_dict["ready"],
+            reason = response_dict["reason"],
+            context = response_dict["context"],
+            event = event
+        )
+
+    def __init__(self, ready, context, reason, event):
+        """
+        Constructor.
+
+        :param ready: Boolean indicating whether or not the service can be used
+        :type ready: bool
+        :param context: A contextual description of the current status: "notice",
+            "success", "error"
+        :type context: str
+        :param reason: A full sentence explanation of the service's status
+        :type reason: str
+        :param event: An event object describing the current state of the service's
+            progress toward readiness
+        :type event: Event
+        """
+        self._ready   = ready
+        self._context = context
+        self._event   = event
+        self._reason  = reason
+
```

Comment:
Can we validate the progress of the event against the `ready` value?

Reply:
Can you elaborate on this hutch? We just accept and show whatever CMC has most recently pushed (which has historically included nonsensical combinations of progress + ready).

Reply:
Could we at least warn on a nonsensical combination of progress + ready so we can start to feed that signal back into the status produces?

Reply:
That makes sense to me. @joannehill this would look like warning whenever `ready` being `True` anytime that progress is present and less than 1.0.

Reply:
@maxhutch would that just involve checking that ready is only `True` when the normalized_progress is `1`?

Reply:
:+1: